### PR TITLE
PEP8 plus optimizations.

### PR DESCRIPTION
**Pull request by warvariuc**
I've built a websocket server based on gevent-websocket and another one using ws4py. The server based on ws4py is quite slow compared to gevent-websocket, so I looked trough ws4py code to see what can be optimized. 

I guess one of the reasons is that ws4py targets python 3 too having additional code paths. 

The main reason for performance difference should be some architectural decisions, but I am not that good to fix it.

### CODE REVIEW

File: ws4py/__init__.py [hunk 1]
```diff
@@ -55,13 +56,3 @@ def configure_logger(stdout=True, filepath=None, level=logging.INFO):
         logger.addHandler(h)
 
     return logger
-
-def format_addresses(ws):
```
Comment (warvariuc) [inline ws4py/__init__.py]: Moved to `WebSocket.__str__`

**Comment by Lawouach** [comment]
Hello,

Well a big thanks for your hard work. I don't have the time going through all of the changes right now but will do my best to do so asap.

Have you witnessed performance gains then?

**Comment by warvariuc** [comment]
You decide if you want to merge them.
No I don't have precise becnhmarks, so I believe the speed if improved did it insignificantly.
Also I not sure I am doing the benchmarks or interpreting the results correctly.
https://github.com/warvariuc/python-websocket-channels/blob/202897fa61068111757ff99e449d047334ed207f/test.py
https://github.com/warvariuc/python-websocket-channels/blob/202897fa61068111757ff99e449d047334ed207f/echo_gevent_server.py

**Comment by warvariuc** [comment]
I don't want to keep the gevent-websocket version, because though it's fast, you cannot plug in control of the handshake -- for example check cookies, session, etc.
https://github.com/warvariuc/python-websocket-channels/blob/202897fa61068111757ff99e449d047334ed207f/websocket_channels.py#L20

**Comment by Lawouach** [comment]
I will see, there might be a few aspects I dont merge (I'm usually not a big fan of local imports for instance, a matter of consistancy with my usual code pratices).

With that said, I don't really know what i'm going to do with ws4py anyway. I feel there are much better alternatives with bigger communities out there. Is there a future for ws4py worth heavily investing in? 

At least, I want the existing code to be as bug-free as possible and I do appreciate it's simple enough for many people.

**Comment by warvariuc** [comment]
> With that said, I don't really know what i'm going to do with ws4py anyway. I feel there are much better alternatives with bigger communities out there. Is there a future for ws4py worth heavily investing in?

What alternatives would you suggest?

I like ws4py because it's quite simple to understand and plug in, because I want to use Redis's pub/sub to implement channels.

**Comment by Lawouach** [comment]
Autobahn (http://crossbar.io/) and Tornado come to mind.

**Comment by warvariuc** [comment]
I hate Twisted :( . I like Gevent :).

**Comment by Lawouach** [comment]
I understand but, to be fair, Autobahn does support asyncio as well these days. Anyway, if you're happy with ws4py, I will not complain. I just wanted to be fair before you invest anymore time. This is a small project with a tiny community :)

**Comment by warvariuc** [comment]
Ok, then myabe my third try will be with Autobahn + asyncio + Python3 :) But I hate camelCase they use everywhere.

**Comment by methane** [comment]
@warvariuc FYI, I've released https://pypi.python.org/pypi/wsaccel to accelerate frame masking.

**Comment by Lawouach** [comment]
I hear you :)

Remember that ws4py does support Python3 and asyncio. It's just bare websocket whereas autobahn has much more to offer (pubsub and all).

**Comment by Lawouach** [comment]
Indeed, wsaccel does improve performances a lot under some cases. See http://www.defuze.org/oss/ws4py/testreports/servers/0.3.4/

**Comment by warvariuc** [comment]
@methane Yes I `pip install`ed it.
@Lawouach Would you make a new release ti `ws4py`? This commit https://github.com/Lawouach/WebSocket-for-Python/commit/1f206bc0f361fab3c74d94052f56137228bc471a is not in PyPi

**Comment by Lawouach** [comment]
Indeed, it's long overdue. I will but not before the end of the week I'm afraid.

**Comment by warvariuc** [comment]
It's not urgent

**Comment by tilgovi** [comment]
@Lawouach please encourage people to contribute and grow the community! I recently switched to ws4py from gevent-websocket. Of the non-autobahn websocket implementations, I think this is the most actively maintained and I've found it easy to use. The thing about using asyncio is that it's not really well used with any popular frameworks yet, afaict. So, unless I want my websocket server to be separate from the rest of my pyramid application, I can't use asyncio. This is the best gevent solution I found that makes it easy for me to have everything in one place.

**Comment by Lawouach** [comment]
Hi, thanks for the kind words regarding ws4py. Its community is small and quiet so it's hard to understand whether or not it's actually useful to anyone :)

**Comment by tito** [comment]
Hi. Thanks you for your work, but i can't merge it like that. It is a little bit outdated, and such changes (like pep8) will also break others PR for minor adjustements.

Also, few changes (like removing the format_address) may also broke for people that use it right now.

Consider rebase your branch if possible :)

**Pull request closed without merging**